code/modules/hydroponics/grown/kudzu.dm

```diff
@@ -57,27 +57,34 @@
 		if(prob(20))
 			mutations.Remove(pick(temp_mut_list))
 		temp_mut_list.Cut()
+
 	if(S.has_reagent("welding_fuel", 5))
 		for(var/datum/spacevine_mutation/SM in mutations)
 			if(SM.quality == POSITIVE)
 				temp_mut_list += SM
 		if(prob(20))
 			mutations.Remove(pick(temp_mut_list))
 		temp_mut_list.Cut()
+
 	if(S.has_reagent("phenol", 5))
 		for(var/datum/spacevine_mutation/SM in mutations)
 			if(SM.quality == MINOR_NEGATIVE)
 				temp_mut_list += SM
 		if(prob(20))
 			mutations.Remove(pick(temp_mut_list))
+		temp_mut_list.Cut()
+
 	if(S.has_reagent("blood", 15))
-		production += rand(15, -5)
+		production = Clamp(production + rand(15, -5),1,10)
+
 	if(S.has_reagent("amatoxin", 5))
-		production += rand(5, -15)
+		production = Clamp(production + rand(5, -15),1,10)
```

Comment:
is production potency or production speed? If it's the latter I have never seen a production speed past 10.